packages/snowpack/src/commands/build.ts

```diff
@@ -223,57 +223,52 @@ export async function command(commandOptions: CommandOptions) {
     process.exit(1);
   }
 
-  const allProxiedFiles = new Set<string>();
+  const allImportProxyFiles = new Set<string>();
   for (const [outLoc, file] of Object.entries(allFilesToResolveImports)) {
     const resolveImportSpecifier = createImportResolver({
       fileLoc: file.locOnDisk!, // we’re confident these are reading from disk because we just read them
       dependencyImportMap: installResult.importMap,
-      isDev: false,
       isBundled,
       config,
     });
     const resolvedCode = await transformFileImports(file, (spec) => {
       // Try to resolve the specifier to a known URL in the project
-      const resolvedImportUrl = resolveImportSpecifier(spec);
-      if (resolvedImportUrl) {
-        // We treat ".proxy.js" files special: we need to make sure that they exist on disk
-        // in the final build, so we mark them to be written to disk at the next step.
-        if (resolvedImportUrl.endsWith('.proxy.js')) {
-          // handle proxied files from web_modules
-          const isWebModule = removeLeadingSlash(resolvedImportUrl).startsWith(
-            removeLeadingSlash(config.buildOptions.webModulesUrl),
+      let resolvedImportUrl = resolveImportSpecifier(spec);
+      if (!resolvedImportUrl) {
+        return spec;
+      }
+
+      // We treat ".proxy.js" files special: we need to make sure that they exist on disk
+      // in the final build, so we mark them to be written to disk at the next step.
+      const isAbsoluteUrlPath = resolvedImportUrl.startsWith('/');
+      const isProxyImport = resolvedImportUrl.endsWith('.proxy.js');
+      if (isProxyImport) {
+        if (isAbsoluteUrlPath) {
+          allImportProxyFiles.add(
+            path.resolve(cwd, config.devOptions.out, removeLeadingSlash(resolvedImportUrl)),
           );
-          if (isWebModule) {
-            allProxiedFiles.add(
-              path.resolve(
-                cwd,
-                config.devOptions.out,
-                removeLeadingSlash(config.buildOptions.webModulesUrl),
-                sanitizePackageName(spec),
-              ),
-            );
-          } else {
-            // handle local proxied files
-            allProxiedFiles.add(
-              resolvedImportUrl.startsWith('/')
-                ? path.resolve(cwd, spec)
-                : path.resolve(path.dirname(outLoc), spec),
-            );
-          }
+        } else {
+          allImportProxyFiles.add(path.resolve(path.dirname(outLoc), resolvedImportUrl));
         }
-        return resolvedImportUrl;
       }
-      return spec;
+
+      // When dealing with an absolute import path, we need to honor the baseUrl
+      if (isAbsoluteUrlPath) {
+        return removeTrailingSlash(config.buildOptions.baseUrl) + resolvedImportUrl;
```

Comment:
from talking with @drwpow: it would be great if this could be relative

Reply:
See https://www.pika.dev/npm/snowpack/discuss/501

Reply:
done!